### PR TITLE
bugfix in select search focus

### LINKED ISSUE #6007: Search select doesn't work on quick typing
### Expected Behavior

Select Search should show that value which is being typed by the user in the input box

### Actual Behavior

When quickly typed, the input box looses focus and a letter (sometimes two letters) gets skipped. This only happens when there is already an option selected

### URL, screen shot, or Codepen exhibiting the issue

When no option is selected. Typing the keyword 'india' shows the same value in the input box.
When some option from the list is selected, typing 'india' very quickly only shows 'idia' in the input box

grommet_search_select_issue

Looks like when the keyword being typed gets matched to any value in the option list (here 'i' matching to option 1, option 2...), the list is re-rendered with new set of options. During that change, the input field looses focus and regains is back after a small delay.

### Steps to Reproduce

1. Got to [search-select story](https://storybook.grommet.io/?path=/story/input-select-search--search)
2. Select any option
3. Type the keyword 'india' very quickly

The same is reproducible in a local project using the `Select` component with similar configuration.

### Your Environment

- Grommet version: 2.21.0
- Browser Name and version: Tested on Firefox 98.0.1 and Chrome 99.0.4844.74
- Operating System and version (desktop or mobile): macOS Big Sur 11.2.1

### PR DESCRIPTION
**Pull request by B-Esmaili**
#### What does this PR do?
Fixes #6007

#### Where should the reviewer start?
`src/js/components/Select/SelectContainer.js`

#### What testing has been done on this PR?

Changes are test using `yarn test` , test resulted in one snapshot test failure. According to changes i have made ,we can update snapshot to match changes. If you confirm it the snapshot could be updated.

#### What are the relevant issues?
https://github.com/grommet/grommet/issues/6007
#### Is this change backwards compatible or is it a breaking change?
As far as i know it's not a breaking change , but it should be double cheked by a reviewer.

### CODE REVIEW

**Review by jcfilben — COMMENTED** [review]
It looks like there are some unrelated changes in this PR. I think the only code we want to change that is related to the issue is this block:
```
useEffect(() => {
  if (activeIndex === -1 && search && optionIndexesInValue.length) {
    setActiveIndex(optionIndexesInValue[0]);
  }
}, [activeIndex, optionIndexesInValue]);
```

**Comment by B-Esmaili** [comment]
> It looks like there are some unrelated changes in this PR. I think the only code we want to change that is related to the issue is this block:
> 
> ```
> useEffect(() => {
>   if (activeIndex === -1 && search && optionIndexesInValue.length) {
>     setActiveIndex(optionIndexesInValue[0]);
>   }
> }, [activeIndex, optionIndexesInValue]);
> ```

What is your suggestion? what changes should we make in this part?

File: src/js/components/Select/SelectContainer.js [hunk 1]
```diff
@@ -97,13 +97,6 @@ const SelectContainer = forwardRef(
         optionsNode.children[activeIndex].focus();
     }, [activeIndex]);
```
Comment (jcfilben) [inline src/js/components/Select/SelectContainer.js:99]: I think we could remove search from this useEffect to solve the issue
```suggestion

// adjust activeIndex when options change
    useEffect(() => {
      if (activeIndex === -1 && optionIndexesInValue.length) {
        setActiveIndex(optionIndexesInValue[0]);
      }
    }, [activeIndex, optionIndexesInValue]);
    
```

Comment (B-Esmaili) [reply src/js/components/Select/SelectContainer.js:99]: This change didn't resolve the issue.

**Comment by B-Esmaili** [comment]
I reverted my previous commit and made another commit [#4590b37](https://github.com/grommet/grommet/pull/6030/commits/4590b37789ea74b35f46a8adaed1fbdc15f2643d) which seems it has resolved the issue.

File: src/js/components/Select/SelectContainer.js (hunk 1 above)
Comment (jcfilben) [reply src/js/components/Select/SelectContainer.js:104]: Oops yes sorry about that I must not have been typing fast enough for the bug to happen when I tested!

Comment (jcfilben) [reply src/js/components/Select/SelectContainer.js:104]: Oops yes sorry about that I must not have been typing fast enough for the bug to happen when I tested!

**Review by jcfilben — APPROVED** [review]
This looks good to me

**Review by amandadupell — APPROVED** [review]
looks good! tested with storybook

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**